src/ui/image_viewer.py

```
@@ -527,10 +565,91 @@ def _on_slider(self, value: int) -> None:
+        # Set the output path for the GIF in the same directory as the input images
+        output_path = image_dir / "animation.gif"
+
+        # Generate the frames for the GIF by normalizing and converting each image to uint8
+        try:
+            frames: List[np.ndarray] = [
+                self._prepare_frame_for_gif(self.handler.get_image_at_index(i))
+                for i in range(image_count)
+            ]
+            # Check if frames were generated successfully
+            if not frames:
+                raise ValueError("No frames available.")
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)